tests/meta-data-from-ast-test.js

```diff
@@ -0,0 +1,186 @@
+'use strict';
+
+const expect = require('chai').expect;
+const metaDataFromAST = require('../lib/meta-data-from-ast');
+const defaultParser = require('../lib/default-parser');
+
+function toAST(source) {
+  return defaultParser(source);
+}
+
+describe('.metaDataFromAST', function() {
+  describe('ES5', function() {
+    const FOO = toAST(`define('foo', ['a', 'b/c'], function() { });`);
+    const A = toAST(`define('a', ['exports', 'require'], function() { })`);
+    const C = toAST(`define('c', ['../a', '../d'], function() { })`);
+    const D = toAST(`define('d', ['foo'], function() { })`);
+
+    it.only('extracts deps', function() {
+      expect(metaDataFromAST(FOO)).to.eql({
+        name: 'foo',
+        imports: {
+          a: ['default'],
+          'b/c': ['default'],
+        },
+        exports: {},
+      });
+
+      expect(metaDataFromAST(A)).to.eql({
+        name: 'a',
+        imports: {
+          exports: ['default'],
+          require: ['default'],
+        },
+        exports: {},
+      });
+
+      expect(metaDataFromAST(C)).to.eql({
+        name: 'c',
+        imports: {
+          '../a': ['default'],
+          '../b': ['default'],
+        },
+        exports: {},
+      });
+
+      expect(metaDataFromAST(D)).to.eql({
+        name: 'd',
+        imports: {
+          foo: ['default'],
+        },
+        exports: {},
+      });
+    });
+  });
+
+  describe('ES6', function() {
+    const FOO = toAST(`
+      import x from 'a';
+      import y from 'b/c';
+    `);
+
+    const A = toAST(``);
+    const C = toAST(`
+      import a from '../a';
+      import d from '../d';
+    `);
+
+    const D = toAST(`import foo from 'foo';`);
+
+    const E = toAST(`
+    export var a = 1;
+    export let b = 1;
+    export const c = 1;
+    export function f () { };
+    export default function () { };
+    export default class C { };
+    `);
+
+    const F = toAST(`
+    export { default } from '../b';
+    export { default as bar } from '../b';
+    export { default as baz } from '../c';
+    export { apple } from '../d';
+    `);
+
+    it('extracts deps', function() {
+      expect(metaDataFromAST(FOO)).to.eql({
+        name: null,
```

Comment:
`name` is `null` because the AST of a es6 module does not know its own name, and at thus isn't applicable.

Reply:
Seems totally fine to require that the thing invoking `metaDataFromAST` set `name` on the returned POJO. 

Also, eventually we should be able to identify the packages own name once the [import.meta proposal](https://github.com/tc39/proposal-import-meta) (which is stage 3) lands...